[EDU-3115] feat: add secure application (#462)
aziontech/docs: src/content/docs/en/pages/secure-journey/secure-application.mdx

---
title: Secure an application
description: Use Edge Firewall to secure your applications.
meta_tags: edge, security, application
namespace: docs_secure_application
menu_namespace: secureMenu
permalink: /documentation/products/secure/secure-application/
---

import Button from '~/components/Button.astro'
import Badge from '~/components/Badge.astro'

> [Edge Firewall](/en/documentation/products/edge-firewall/) allows you to secure your [edge applications](/en/documentation/products/guides/build/build-an-application/) and [domains](/en/documentation/products/guides/configure-a-domain/) against several types of threats and attacks.

Reviewer: Achei muitos links para um texto bem curto...
talvez tirar o de EF deixa melhor?

> [Edge Firewall](/en/documentation/products/edge-firewall/) allows you to secure your [edge applications](/en/documentation/products/guides/build/build-an-application/) and [domains](/en/documentation/products/guides/configure-a-domain/) against several types of threats and attacks.

Author: ```suggestion
**Edge Firewall** allows you to secure your [edge applications](/en/documentation/products/guides/build/build-an-application/) and [domains](/en/documentation/products/guides/configure-a-domain/) against several types of threats and attacks.
```

Azion provides a range of tools to configure your edge firewall:

- [Real-Time Manager (RTM)](https://manager.azion.com/): provides a user-friendly interface to configure your edge firewall and rule sets.
- [Azion API](https://api.azion.com/#00268e07-463f-4de1-9b45-ceeab187bf19): allows you to create, configure, and automate edge firewalls and its related modules and features.

> - [Azion SDK](): allows you to improve the development of your Go application.

Reviewer: ```suggestion
- [Azion SDK](/en/documentation/devtools/sdk/go/): allows you to improve the development of your Go application.
```

- [Terraform](/en/documentation/products/secure/automate/terraform/): provides integration for your infrastructure with Azion using Terraform. Configurations available for main settings and edge functions instances.

> - [Edge Runtime](): leverages APIs for networking and data handling.

Reviewer: ```suggestion
- [Edge Runtime](/en/documentation/products/edge-application/edge-functions/runtime/overview/): leverages APIs for networking and data handling.
```

---

## Step 1: Configure main settings

To create an edge firewall, the first step is to provide:

- A unique and easy-to-remember name.
- The domains you want to associate with the edge firewall.
- The modules you want to use.

>     - This is an optional step, but to fully secure your applications, it's recommend to enable all available modules.

Reviewer 2: ```suggestion
    - This is an optional step, but to fully secure your applications, it's recommended to enable all available modules.
```

Optionally, you can also enable the **Debug Rules** feature.

<Button href="/en/documentation/products/guides/secure/configure-main-settings/" text="go to configure main settings guide" variant="secondary"></Button>

---

## Step 2: Choose a function to instantiate

The second step is to instantiate an edge function.

Requires:

<Badge variant="accent">[Edge Functions for Edge Firewall](/en/documentation/products/edge-firewall/edge-functions/firewall/)</Badge>

> Instantiating an existing edge function created for security purposes, such as the **Bot Protection** or **Bot Management** integrations, reinforces your security policies and adds another layer of protection for your applications.

Reviewer: ```suggestion
Instantiating an existing edge function created for security purposes, such as the **Bot Protection** or **Scheduled Blocking** integrations, reinforces your security policies and adds another layer of protection for your applications.
```
